### ISSUE TITLE
`.meta.client.transact_write_items` has incorrect type

### ISSUE THREAD

**Issue by magicmark** [youtype/mypy_boto3_builder#164]
**Describe the bug**

`dynamodb.meta.client.transact_write_items` appears to want to use `AttributeValueTypeDef` types, when runtime is ok using non-marshalled types

**To Reproduce**

Here's a sample script
```python
import boto3
dynamodb = boto3.resource("dynamodb", region_name="us-east-1", endpoint_url="http://localhost:23212")
dynamodb.meta.client.transact_write_items(
    TransactItems=[
        {
            "Update": {
                "TableName": "foo",
                "Key": {"bar": "bar"},
                "UpdateExpression": "set baz = :baz",
                "ExpressionAttributeValues": {":baz": "baz"},
            }
        }
    ]
)
```

Using this schema:
<details>

```bash
 aws dynamodb create-table \
    --endpoint-url "http://localhost:23212" \
    --table-name foo \
    --attribute-definitions \
        AttributeName=bar,AttributeType=S \
    --key-schema \
        AttributeName=bar,KeyType=HASH \
    --provisioned-throughput \
        ReadCapacityUnits=5,WriteCapacityUnits=5 \
    --table-class STANDARD
```

</details>

**This works at runtime!**

But here's the error when we run mypy:

```
$ mypy test.py
test.py:8:25: error: Dict entry 0 has incompatible type "str": "str"; expected "str": "AttributeValueTypeDef"
test.py:10:47: error: Dict entry 0 has incompatible type "str": "str"; expected "str": "AttributeValueTypeDef"
Found 2 errors in 1 file (checked 1 source file)
```

Uh oh! Well fair enough, we are doing `meta.client` so maybe we're supposed to write these as marashalled values like we would in the client client?

```diff
 import boto3
 dynamodb = boto3.resource("dynamodb", region_name="us-east-1", endpoint_url="http://localhost:23212")
 dynamodb.meta.client.transact_write_items(
     TransactItems=[
         {
             "Update": {
                 "TableName": "foo",
-                "Key": {"bar": "bar"},
+                "Key": {"bar": {"S": "bar" }},
                 "UpdateExpression": "set baz = :baz",
-                "ExpressionAttributeValues": {":baz": "baz"},
+                "ExpressionAttributeValues": {":baz": {"S": "baz"}},
             }
         }
     ]
 )
```

This makes mypy happy, but makes runtime sad:

```
$ test.py
Traceback (most recent call last):
  File "test.py", line 10, in <module>
    "ExpressionAttributeValues": {":baz": {"S": "baz"}},
  File "<project>/venv/lib/python3.7/site-packages/botocore/client.py", line 508, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "<project>/venv/lib/python3.7/site-packages/botocore/client.py", line 915, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (ValidationException) when calling the TransactWriteItems operation: Invalid attribute value type
```

**Additional context**
```bash
$ uname -rs
Linux 5.4.0-1061-aws

$ pip freeze | grep boto
boto==2.49.0
boto3==1.24.51
boto3-stubs==1.24.53
botocore==1.27.51
botocore-stubs==1.27.42.post3
mypy-boto3-dynamodb==1.24.36.post1
mypy-boto3-s3==1.24.36.post1
mypy-boto3-sts==1.24.36.post1
```

(Is this related to https://github.com/youtype/mypy_boto3_builder/issues/148 ?)

**Comment by magicmark** [comment]
(Explicit types doesn't seem to help either)

```python
import boto3
from mypy_boto3_dynamodb.service_resource import DynamoDBServiceResource
dynamodb: DynamoDBServiceResource = boto3.resource("dynamodb", region_name="us-east-1", endpoint_url="http://localhost:23212")
```

produces the same mypy error

**Comment by vemel** [comment]
Thank you for the report!

Looks like this function uses `boto3` attribute schema rather than raw `botocore`. I will release a fix shortly.

**Comment by vemel** [comment]
Looks like an error in `boto3`/`botocore` shapes. 

As you see, official documentation also uses raw `botocore` attribute schema: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Client.transact_write_items

Probably, the best way to fix this is to accept both `boto3` and `botocore` schemas for all client methods.

**Comment by vemel** [comment]
Fixed in `mypy-boto3-dynamodb 1.24.55.post1`. Please let me know if it works as it should.

**Comment by magicmark** [comment]
Wow - fixed in less than a day - outstanding, thank you!

<3 <3
